rmgpy/solver/surface.pyx

```diff
@@ -384,17 +403,31 @@ cdef class SurfaceReactor(ReactionSystem):
 
         C = np.zeros_like(self.core_species_concentrations)
         V = self.V  # constant volume reactor
+        A = self.V * surface_volume_ratio_si  # area
+        total_sites = self.surface_site_density.value_si * A  # todo: double check units
 
+        current_surface_coverages = {}
         for j in range(num_core_species):
             if species_on_surface[j]:
                 C[j] = (N[j] / V) / surface_volume_ratio_si
+                surface_site_fraction = N[j] / total_sites
+                smiles = cov_dep_index_smiles[j]
+                current_surface_coverages[smiles] = surface_site_fraction
             else:
                 C[j] = N[j] / V
             #: surface species are in mol/m2, gas phase are in mol/m3
             core_species_concentrations[j] = C[j]
 
         for j in range(ir.shape[0]):
             k = kf[j]
+            if j in cov_dep:
+                cov_dep_j = cov_dep[j]
+                for smiles, parameters in cov_dep_j.items():
+                    if smiles in current_surface_coverages:
+                        theta = current_surface_coverages[smiles]
+                        if theta >= 0.1:
```

Comment:
This is arbitrary. We just didn't want to have the rates do funky things at very low coverages, so we decided to not use the correction if the coverage was lower. I think when we plotted the line out, around .1 was the global minimum before the coverage started going to infinity

Reply:
Shouldn't all the expressions go to "no effect" when coverage goes to zero? like you add zero, or multiply by something to the power zero, etc.

Reply:
This is somewhat arbitrary.  `* theta ** parameters['m']`  term is undefined at 0 coverage and m<0, so I think we need to set some lower bound here as we may run in to numerical issues if theta is very very small.  I think we chose .1 because the lowest coverages that cov dep DFT studies look at are around this coverage, but maybe the .1 threshold here is too high.

Reply:
I hadn't considered someone using a negative _m_. That seems like a bad idea. Maybe that's the problem.

Reply:
I still don't think coverage effects should suddenly kick in at 10% coverage. 

I now understand that people sometimes use negative m.
The use case of a vacant site having _m=-1_ so that a dissociative adsorption rate is first order wrt vacancies, is then going to be multiplied by vacancy fraction squared, so should cancel out as long as we avoid floating point overflows. Maybe use a threshold of `1e-15` ?

Reply:
ok, will change from 0.1 to 1e-15